collectors/monitors.go

```diff
@@ -371,6 +448,29 @@ func (m *MonitorCollector) collect() error {
 
 	m.NodesinQuorum.Set(float64(len(stats.Quorum)))
 
+	// Ceph versions, one loop for each daemon.
+	// In a consistent cluster, there will only be one iteration (and label set) per daemon.
+	for daemon, vers := range versions {
+		for version, num := range vers {
+			// We have a version, which is something like the following, how we want to map it
+			// ceph version 12.2.13-30-aabbccdd (c5b1fd521188ddcdedcf6f98ae0e6a02286042f2) luminous (stable)
+			//              version_tag          sha1                                      release_tag
+			res := versionRegexp.FindStringSubmatch(version)
+			if len(res) != 4 {
+				continue
+			}
+
+			m.CephVersions.WithLabelValues(daemon, res[1], res[2], res[3]).Set(float64(num))
```

Comment:
For clarity you could use https://golang.org/pkg/regexp/#Regexp.SubexpIndex here, but it also makes this much wordier.